tests/test_writer.py

```diff
@@ -823,3 +824,115 @@ def test_unspecified_images_keys(self):
+        ),
+        ids=["5D", "3D", "2D"],
+    )
+    def shape(self, request):
+        return request.param
+
+    @pytest.fixture(params=[True, False], ids=["scale", "noop"])
+    def scaler(self, request):
+        if request.param:
+            return Scaler()
+        else:
+            return None
+
+    @pytest.mark.parametrize(
+        "format_version",
+        (
+            pytest.param(FormatV01, id="V01"),
+            pytest.param(FormatV02, id="V02"),
+            pytest.param(FormatV03, id="V03"),
+            pytest.param(FormatV04, id="V04"),
+        ),
+    )
+    def test_multiscale_label_writer(self, shape, scaler, format_version):
+
+        version = format_version()
+        axes = "tczyx"[-len(shape) :]
+        transformations = []
+        for dataset_transfs in TRANSFORMATIONS:
+            transf = dataset_transfs[0]
+            # e.g. slice [1, 1, z, x, y] -> [z, x, y] for 3D
+            transformations.append(
+                [{"type": "scale", "scale": transf["scale"][-len(shape) :]}]
+            )
+
+        # create the actual label data
+        label_data = np.random.randint(0, 1000, size=shape)
+        if version.version in ("0.1", "0.2"):
+            # v0.1 and v0.2 require 5d
+            expand_dims = (np.s_[None],) * (5 - len(shape))
+            label_data = label_data[expand_dims]
+            assert label_data.ndim == 5
+        label_name = "my-labels"
+        if scaler is None:
+            transformations = [transformations[0]]
+            labels_mip = [label_data]
+        else:
+            labels_mip = scaler.nearest(label_data)
+
+        # create the root level image data
+        self.create_image_data(shape, scaler, version, axes, transformations)
+
+        write_multiscale_labels(
+            labels_mip,
+            self.root,
+            name=label_name,
+            fmt=version,
+            axes=axes,
+            coordinate_transformations=transformations,
+        )
+
+        # Verify image data
+        reader = Reader(parse_url(f"{self.path}/labels/{label_name}"))
+        node = list(reader())[0]
+        assert Multiscales.matches(node.zarr)
+        if version.version in ("0.1", "0.2"):
+            # v0.1 and v0.2 MUST be 5D
+            assert node.data[0].ndim == 5
+        else:
+            assert node.data[0].shape == shape
+        print("node.metadata", node.metadata)
+        if version.version not in ("0.1", "0.2", "0.3"):
+            for transf, expected in zip(
+                node.metadata["coordinateTransformations"], transformations
+            ):
+                assert transf == expected
+            assert len(node.metadata["coordinateTransformations"]) == len(node.data)
+        assert np.allclose(label_data, node.data[0][...].compute())
+
+        # Verify label metadata
+        label_root = zarr.open(f"{self.path}/labels", "r")
+        assert "labels" in label_root.attrs
+        assert label_name in label_root.attrs["labels"]
```

Comment:
Should we also ensure that `label_name` is correctly added to the `multiscales` metadata ?